test/e2e/user_workload_monitoring_test.go

```diff
@@ -802,17 +803,41 @@ func assertPrometheusAlertmanagerInUserNamespace(t *testing.T) {
 }
 
 func assertGRPCTLSRotation(t *testing.T) {
+	countGRPCSecrets := func(ns string) int {
+		t.Helper()
+		var result int
+		err := framework.Poll(5*time.Second, time.Minute, func() error {
+			s, err := f.KubeClient.CoreV1().Secrets(ns).List(context.TODO(), metav1.ListOptions{LabelSelector: "monitoring.openshift.io/hash"})
+			if err != nil {
+				return err
+			}
+
+			for _, s := range s.Items {
+				if strings.Contains(s.Name, "grpc-tls") {
+					result++
+				}
+			}
+
+			return nil
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		return result
+	}
+
 	s, err := f.OperatorClient.WaitForSecret(&v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "grpc-tls",
 			Namespace: f.Ns,
 		},
 	})
-
 	if err != nil {
 		t.Fatalf("error waiting for grpc-tls secret: %v", err)
 	}
 
+	expected := countGRPCSecrets(f.Ns) + countGRPCSecrets(f.UserWorkloadMonitoringNs)
```

Comment:
I think this calculation of expected grpc secret count is racy. The test code here waits until the central `grpc-tls` secret is created. However at this point of time there is no guarantee that all subsequent grpc secrets in other namespaces are also already created as their creation is orchestrated in separate task executions which happen concurrently.

Instead, I propose to change this code here to have a const number of five (5) expected secrets, because we know them in forehand, as they are:
1. `openshift-monitoring/grpc-tls`
2. `openshift-monitoring/prometheus-k8s-grpc-tls-[hash]`
3. `openshift-user-workload-monitoring/prometheus-user-workload-grpc-tls-[hash]`
4. `openshift-monitoring/thanos-querier-grpc-tls-[hash]`
5. `openshift-user-workload-monitoring/thanos-ruler-grpc-tls-[hash]`

@openshift/openshift-team-monitoring  wdyt?

Reply:
Hardcoding value here seems fine to me, but I would expect an in-code comment about this.

Reply:
@paulfantom added comment in addition to the commit message, ptal.